binding-service-impl/pom.xml

```diff
@@ -29,5 +29,11 @@
 			<artifactId>esignet-core</artifactId>
 			<version>${esignet.core.version}</version>
 		</dependency>
-	</dependencies>
+        <dependency>
+            <groupId>io.mosip.esignet</groupId>
+            <artifactId>oidc-service-impl</artifactId>
+            <version>${project.version}</version>
+            <scope>compile</scope>
+        </dependency>
```

Comment:
If CacheUtilService is the only dependent service class, we should move CacheUtilService to esignet-core instead of adding oidc-service-impl as a dependency to binding-service-impl. It's important to keep them as separate services to maintain the "separation of concerns."

Reply:
Also, can we limit this PR only to captcha validation? API rate limit can be a separate PR. Let us explore bucket4j for the API rate limit.

Reply:
Limited this PR to captcha validations. API rate limit code is reverted and will be added in separate PR.